python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -0,0 +1,113 @@
+"""Process dosage matrix, GWAS summary statistics, and LD maps for PRS C+T calculation."""
+
+import glob
+import logging
+import re
+import pandas as pd
+from pyarrow import feather  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+def _load_preprocessed_dosage_matrix(dosage_matrix_path: str) -> pd.DataFrame:
+    """Load in dosage matrix generated from bystro-vcf."""
+    return feather.read_feather(dosage_matrix_path)
+
+
+def _load_association_scores(gwas_scores_path: str) -> pd.DataFrame:
+    """Load in GWAS summary statistics provided by user."""
+    return pd.read_csv(gwas_scores_path, delimiter="\t", comment="#")
+
+
+def _load_genetic_maps_from_feather(map_directory_path: str) -> dict[str, pd.DataFrame]:
+    """Load genetic maps from Feather files in the specified directory, using logging for messages.
+
+    Args:
+    ----
+    map_directory_path: The path to the directory containing Feather files.
+
+    Returns:
+    -------
+    A dictionary where keys are 'GeneticMap{chromosome_number}' and values are the corresponding
+    DataFrames loaded from Feather files.
+    """
+    file_pattern = f"{map_directory_path}/*.feather"
+    map_file_list = glob.glob(file_pattern)
+    genetic_maps = {}
+    for file in map_file_list:
+        try:
+            genetic_map = pd.read_feather(file)
+            match = re.search(r"chromosome_(\d+)_genetic_map", file)
```

Comment:
If we can't determine the chromosome, is the file useful?

If it isn't, which I don't think it can be, 1) you have to raise an exception if you don't have a match, rather than logging a warning, 2) you shouldn't load the file until you have successfully parsed the match.